analyzer/UsedAssemblyReferencesDumper.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace ReferenceTrimmer
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class UsedAssemblyReferencesDumper : DiagnosticAnalyzer
+    {
+        internal static readonly string Title = "Unused references should be removed";
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor("RT0001", Title, "'{0}'", "References", DiagnosticSeverity.Warning, true, customTags: WellKnownDiagnosticTags.Unnecessary);

```

Comment:
This rule actually needed? Seems like we only need the side-effect and never actually raise a rule violation

Reply:
It is not effectively needed. However, if the `SupportedDiagnostics` returns an empty array or the `DiagnosticSeverity` is not warning, then the analyzer does not kick in.